cli/src/merge_tools/builtin.rs

```diff
@@ -648,6 +669,7 @@ pub fn edit_merge_builtin(
             .map(|file| file.repo_path.clone())
             .collect_vec(),
         &state.files,
+        |path| tree.path_value(path),
```

Comment:
Can you check the behavior of "merge" (or `jj resolve`) editor?

If returning `tree.path_value(path)` is correct, the callback function shouldn't be named as `select_left()`. It's `select_original` or something. If this function should return the left tree, the same TODO comment as `select_right` would apply.

Reply:
> If I'm not mistaken, `select_right()` also does not seem to be called in the merge editor, but for different reasons: conflicted binary files are currently not rendered as `Binary` sections.

Right. I think I added TODO comment about that. It seemed wrong to split hunks without checking if the content has null character. And I have no idea how left/right binary contents would be selected with the current UI.

Can you insert a comment to `|path| tree.path_value(path)` of `edit_merge_builtin()`? With the current state, it looks `select_right` is wrong whereas `select_left` is correct.

Reply:
I've turned it into a `FIXME`, given that we're pretty sure that the `|path| tree.path_value(path)` implementation is wrong for `select_right`. I've considered replacing it with `unreachable!()`, but that might be a little too hot.

> And I have no idea how left/right binary contents would be selected with the current UI.

I think it might be possible to use the current scm_record UI to select between the "ours" or "theirs" sides to the conflict if both are represented by separate `Binary` sections. It would then be possible to obtain sensible descriptions for either one from the `Merge`, too. Obviously, it wouldn't make sense to select both sections, so scm_record would need to treat multiple `Binary` sections as mutable exclusive.